devtools/maven/src/main/java/io/quarkus/maven/DevMojo.java

```diff
@@ -108,6 +109,7 @@
  * You can use this dev mode in a remote container environment with {@code remote-dev}.
  */
 @Mojo(name = "dev", defaultPhase = LifecyclePhase.PREPARE_PACKAGE, requiresDependencyResolution = ResolutionScope.TEST, threadSafe = true)
+@Execute(phase = LifecyclePhase.PROCESS_CLASSES)
```

Comment:
I have yet to test this, but theoretically `mvn compile quarkus:dev` will now probably run compile twice:
- once because of `compile`
- and once because of the forked lifecycle phase

That's what I meant with you cannot really opt out of `@Execute`.
Maven 4 will be more clever in that regard: https://issues.apache.org/jira/browse/MNG-6566
But until then this might be an issue we don't want to live with.

The only way out of this (AFAICS) is to offer two variants, one forking and one non-forking.
"...-no-fork" is usually the goal name pattern for find for the non-forking variant in varions plugins (which MNG-6566 tries to overcome).

Another (ugly) compensation might be to (try to) detect that we are in a forked execution and skip everything in case we also detect that originally compile was already executted.

Two more things:
- If we offer a variant with `@Execute`, it should not contain anymore the execution of specific goals.
- I think the "test" goal/mojo has to be adjusted as well.

Reply:
I'm not comfortable with changing the way we are building things in 2.7.

Can't we just change the fact that we are executing `compile`?

Reply:
> I'm not comfortable with changing the way we are building things in 2.7.

2.7 specifically or 2.x in general, making it a 3.x candidate?

> Can't we just change the fact that we are executing `compile`?

Sure. But with `@Execute` we would never miss a user defined execution (we had a few issues in that area).

Reply:
At the beginning I was thinking of doing something similar to what we do with the ˋcompile` goal. I think we can stick with it. I will update the branch

Reply:
> 2.7 specifically or 2.x in general, making it a 3.x candidate?

2.7 in particular.

Reply:
So, no `@Execute` for this PR then, resolving this discussion.